content/articles/creating-an-animated-javafx-field-validation-alert/index.md

```diff
@@ -0,0 +1,761 @@
+
+At the end of the article, you will have done the following:
+
+- Create a JavaFX login form
+- Apply styling to the application
+- Create an input validation method
+- Add effects based on the outcomes of the validation
+- Add animations on the controls
+
+### Pre-requisites
+
+The basics needed to follow up with the tutorial quickly include:
+
+- Must be familiar with Java language
+- Java SDK installed on the machine
+- A JavaFX IDE is installed on the machine.
+  I recommend using the latest IntelliJ IDEA Community version. It is free and enables developers to quickly create a JavaFX project.
+  Find it in this [link](https://www.jetbrains.com/idea/download).
+  The latest Ultimate version also supports JavaFX projects.
+- A good internet connection. This will be used in installing additional dependencies.
+
+Once all the above pre-requisites are met, follow the steps below to accomplish the goals:
+
+> The Screenshot images in the article are of the IntelliJ Ultimate edition `2021.2.2` version
+
+### Create a new project
+
+- Open the IDE and click on create a new project. Under the project type options, select JavaFX.
+  Set the following on the right-hand side window:
+
+- `Name`: Login
+- `Group`: authentication
+- `Artifact`: Login
+- `Project SDK`: You can set the latest version or that which you are using.
+
+These are well shown in the image below:
+
+![new javafx project](/engineering-education/creating-an-animated-javafx-field-validation-alert/new-javafx-project.png "New-JavaFX project")
+
+- Click on the `Next` button. On the dependencies window, select the `BootstrapFX`, `ControlsFX`, and `FormsFX` checkboxes.
+  BootstrapFX helps in styling the form using JavaFX styling. The styling is somehow similar to CSS styling.
+  ControlsFX is for allowing controls to be available in the project.
+  These include TextFields, PasswordFields, Labels, DatePickers, and many others. FormFX allows one to easily design forms.
+
+Look at the image below to see the above steps:
+
+![Dependencies](/engineering-education/creating-an-animated-javafx-field-validation-alert/dependencies.png "dependencies")
+
+The newly created project has the following folder structure:
+
+#### Folder Structure
+
+```shell
+.
+├── src
+│   ├── main
+│       ├── java
+│          └── authentication
+│             └── login
+│                 ├── HelloApplication.java
+│                 └── HelloController.java
+│       └── resources
+│          └── authentication
+│             └── login
+│                 └── hello-view.fxml
+├── Login.iml
+└── pom.xml
+```
+
+- Open the `hello-view.fxml` file which contains the contents of the page to be displayed.
+  Near the status bar, there is a button adjacent to the Text button.
+  Click on it to see the FXML file visually using Scene Builder.
+
+  If it displays the `Failed to open the file in Scene Builder` error, click on `Download SceneBuilder`.
+  Click also on the `Download JavaFX` if it also brings an additional Pop-up.

```

Comment:
Use bullets `-` when describing a procedure.

Reply:
Got it...and done!

Reply:
Okay